### ISSUE TITLE
Can’t update 2.9.1 to newer

### ISSUE THREAD

**Issue by edelmaca** [mongoose-os-apps/shelly-homekit#775]
I tried to update one Shelly of mine since a few firmware versions but every version bigger than 2.9.1 was not installed on this one Shelly. My other Shelly had no problem with any of the updates. 

I tried updating with the check and install buttons and by uploading the file and install it afterwards. But I had no luck so far. 

Had the log file started as quick as possible after starting the update procedure.

_96542864 shelly_debug.cpp:227    No log file, sending new entries
96578261 mg_rpc.c:305            Shelly.GetInfoExt via WS_in 192.168.1.174:57492
100974627 shelly_main.cpp:803     Starting firmware update
100986859 esp_ota_backend.c:185   Slot 0, FW: switch1.bin -> 0x8000, FS fs.bin -> 0xbb000
100999472 mgos_ota_core.c:505     0.18% total, esp_init_data_default_v08.bin 128 of 128
101133102 esp_ota_backend.c:338   Start writing fs.bin (262144) @ 0xbb000
101181125 mgos_ota_core.c:505     0.24% total, fs.bin 512 of 262144
101305695 shelly_main.cpp:543     Up 101.30, HAP 0/0/12 ns 0, RAM: 20784/38760; st 0; disabled
104928968 esp_ota_backend.c:376   Write finished, checksum ok
104947798 mgos_ota_core.c:505     27.45% total, rboot.bin 512 of 2320
105290420 esp_ota_backend.c:334   Skip writing switch1.bin (696688) @ 0x8000 (digest matches)
105307091 mgos_ota_core.c:505     27.70% total, switch1.bin 512 of 696688
106955299 mgos_ota_core.c:505     54.90% total, switch1.bin 262656 of 696688
108612886 mgos_ota_core.c:505     82.10% total, switch1.bin 524800 of 696688
109307653 shelly_main.cpp:543     Up 109.30, HAP 0/0/12 ns 0, RAM: 22188/38760; st 0; disabled
109666044 mgos_ota_core.c:637     Reached the end of archive
109672110 esp_ota_backend.c:408   Resetting RF cal data @ 0x1fb000
109753297 esp_ota_backend.c:438   New rboot config: prev_rom: 1, current_rom: 0 current_rom addr: 0x8000, current_rom size: 696688, current_fs addr: 0xbb000, current_fs size: 262144
109761010 mgos_ota_core.c:818     Update finished, result 1 (Update applied, rebooting)
109773671 mgos_ota_core.c:861     Update requires reboot
109777911 mgos_system.c:58        Rebooting in 500 ms_

The file looks good for me. It says the update was finished. But the Shelly webpage says otherwise. It sits on 2.9.1 since the release of this version. 🤷🏻‍♂️
Anyone with the same problem or some tips or help? I would appreciate it. 
Will help anyone who want to try to get to the cause of this Problem.

**Comment by rojer** [comment]
indeed, the update process was successful and the file is ok.
unchanged version means that the new firmware failed to fully initialize, crashed and was rolled back. it's a safety mechanism to prevent bricking the device.
the question is - why does the new firmware fail and device reboots. unfortunately, it's not possible to tell without observing the logs on the serial port.
can you think of anything special about this shelly's configuration compared to others?
can you try resetting it and updating without any other changes (except wifi)?

**Comment by edelmaca** [comment]
Thanks for diggin in.
My Shelly's are set up exactly the same. So no, can't think of any differences.
If I reset the Shelly, will it present itself as a new device and I have to redo any automations and functions setup with this device?

**Comment by rojer** [comment]
i'm afraid yes, you will have to re-add it and re-do autiomations

**Comment by edelmaca** [comment]
Hm, so it looks like I will stay with this firmware release for a while.
Im moving sometime next year and have to redo my HomeKit Home. So this is the time I will reset it then. Now it's a  working running system...I take the phrase "never change a running system" serious for now.
Thank you for taking the time to help me.

**Comment by edelmaca** [comment]
Ok. Strange
I just checked for updates and I have updated my Shelly's to 2.10.0. No problems for my "working" Shelly and guess what...the one who got stuck on 2.9.1, which had problems with every 2.9.x update, updated like it had never done anything strange before. Rebooted it to check the firmware version and it looks like it got through and is now on the newest version 2.10.0 as well. 

Case closed. 
What was the problem? No clue. 🤷🏻‍♂️
